libc-top-half/musl/src/thread/pthread_condattr_setclock.c

```diff
@@ -1,9 +1,11 @@
 #include "pthread_impl.h"
 
+#include <common/clock.h>
+
 int pthread_condattr_setclock(pthread_condattr_t *a, clockid_t clk)
 {
-	if (clk < 0 || clk-2U < 2) return EINVAL;
+	if (clk->id < 0 || clk->id-2U < 2) return EINVAL;
```

Comment:
I wonder if we can drop `cloudlibc/src/common/clock.h` completely and just use __wasi_clockid_t as the definition for clockid_t?.. avoiding the need for this change?

Reply:
Actually I guess the comments in `libc-bottom-half/cloudlibc/src/common/clock.h` are pretty clear this was a deliberate choice.   I guess this change is needed.

Reply:
ok